### ISSUE TITLE
Direct link to file?

### ISSUE THREAD

**Issue by javilumbrales** [torre76/django-googledrive-storage#30]
Hi, first of all I'd like to thank you for the great work. I've just tried this and worked straight away, all perfect, file gets uploaded and so. Now, the only issue I'm having is that when I click on the link from the admin, or if I use django-restful to serialize a model using this storage, the link exposed doesn't take to the actual "download" of the file, but to a sort of preview page, in which you can see the file if it's an image or download the file, etc. Is there a way to get the direct link to the content, skipping google's "middle" page? I was hopping to integrate this with an android app to actually download the uploaded file, but this would download the html page instead of the actual file.

Any help on this would be greatly appreciated.

Thank you!

**Comment by torre76** [comment]
Hi @javilumbrales,

sorry for being late.
When you says you obtain the link of the file, do you referee to the 

`def url(self, name)`

of the storage?

The link provided by this method has to be (as per storage spec) _an absolute URL where the file's contents can be accessed directly by a Web browser_ and the middle link of Google Drive perfectly suits this definition.

Having said that, you could make a _dirty_ trick and invoke private method of the storage to obtain the link you want, specifically, if you have the absolute path of the file as _name_ the code you are looking for is:

```python
from gdstorage.storage import GoogleDriveStorage

gd_storage = GoogleDriveStorage()
name = "/folder/file_on_google.drive"

direct_url = gd_storage.self._check_file_exists(name)["downloadUrl"]
```

If you use an IDE like _PyCharm_ this code will be highlighted with a warning because you are using a private method to achieve the result you want, but it should work.

**Comment by javilumbrales** [comment]
Hi,
Thank you very much for your response. Yes, I meant the url method.

I tried something like that and for some reason didn't seem to work, but anyway eventually, I managed to workaround it creating a proxy/view in the server like this:

    def show_picture(self, id):
        image = get_object_or_404(Picture, pk=id)
        filename = image.image.file.name.split('/')[-1]
        response = HttpResponse(image.image.file, content_type='image/jpeg')

        return response

Basically, I open the file and output it as response, which is not ideal but good enough for what I need.

Once again, thank you very much.

Best regards,
Javier

**Comment by torre76** [comment]
It would not work because the `downloadURL` is the raw stream of the data without header and identification, from there you should open a `IOStream` and read the data and then add the headers that identifies the content.

More or less what you have done with the proxy method.

Sorry for being late, but you have created a good workaround.

Regards
Gian Luca
